core/src/main/java/eu/fasten/core/data/JSONUtils.java

```diff
@@ -0,0 +1,287 @@
+package eu.fasten.core.data;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JSONUtils {
+    /**
+     * Converts an {@link ExtendedRevisionJavaCallGraph} object to its corresponding JSON String
+     * without any object creation in between. It creates a {@link StringBuilder) in the beginning
+     * and only appends to it in order to decrease the memory and time consumption.
+     *
+     * @param erjcg and object of java revision call graph to be converted to JSON String.
+     * @return the corresponding JSON String.
+     */
+    public static String toJSONString(final ExtendedRevisionJavaCallGraph erjcg) {
+        var result = new StringBuilder("{");
+        appendArtifactInformation(erjcg, result);
+        appendCha(result, erjcg.classHierarchy);
+        appendGraph(result, erjcg.graph);
+        if (erjcg.timestamp >= 0) {
+            appendKeyValue(result, "timestamp", erjcg.timestamp, true);
+        }
+        result.append("}");
+        return result.toString();
+    }
+
+    /**
+     * Appends general information of the revision to the beginning of the StringBuilder.
+     *
+     * @param erjcg  the object to extract the information from.
+     * @param result the StringBuilder to append the information.
+     */
+    private static void appendArtifactInformation(ExtendedRevisionJavaCallGraph erjcg,
```

Comment:
I would suggest to have the `StringBuilder` as the first parameter for coherency with other methods.

Reply:
fixed b98c359